File: mvps/php/app/Http/Controllers/UserController.php
<?php

namespace App\Http\Controllers;

use App\Http\Requests\CreateUserRequest;
use App\Http\Requests\ListUsersRequest;
use App\Services\UserService;
use Illuminate\Http\JsonResponse;

class UserController extends Controller
{
    public function __construct(
        private readonly UserService $users,
    ) {}

    public function store(CreateUserRequest $request): JsonResponse
    {
        $user = $this->users->create($request->validated('name'));

        return response()->json($user, 201);
    }

    public function index(ListUsersRequest $request): JsonResponse
    {
        $data = $request->validated();

        return response()->json($this->users->list(
            (int) $data['page'],
            (int) $data['page_size'],
            $data['sort'],
            $data['order'],
        ));
    }
}

File: java/src/main/java/com/meetup/beervote/web/UserController.java
package com.meetup.beervote.web;

import com.meetup.beervote.dto.ApiDtos.CreateUserRequest;
import com.meetup.beervote.dto.ApiDtos.UserListResponse;
import com.meetup.beervote.dto.ApiDtos.UserResponse;
import com.meetup.beervote.user.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/users")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public UserResponse create(@RequestBody CreateUserRequest request) {
        return userService.create(request);
    }

    @GetMapping
    public UserListResponse list(
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(name = "page_size", defaultValue = "20") int pageSize,
            @RequestParam(defaultValue = "created_at") String sort,
            @RequestParam(defaultValue = "desc") String order
    ) {
        return userService.list(page, pageSize, sort, order);
    }
}
